src/txn/txn_ckpt.c

```diff
@@ -906,12 +907,15 @@ __txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[])
         WT_ERR(ret);
 
         /*
-         * Once the history store checkpoint is complete, we increment the checkpoint generation of
-         * the associated b-tree. The checkpoint generation controls whether we include the
-         * checkpoint transaction in our calculations of the pinned and oldest_ids for a given
-         * btree. We increment it here to ensure that the visibility checks performed on updates in
-         * the history store do not include the checkpoint transaction.
+         * Once the history store checkpoint is complete, we update the statistic with the most
+         * recent history store file size on disk, and increment the checkpoint generation of the
+         * associated b-tree. The checkpoint generation controls whether we include the checkpoint
+         * transaction in our calculations of the pinned and oldest_ids for a given btree. We
+         * increment it here to ensure that the visibility checks performed on updates in the
+         * history store do not include the checkpoint transaction.
          */
+        WT_ERR(__wt_block_manager_named_size(session, WT_HS_FILE, &hs_size));
```

Comment:
We should do this after we sync the hs file in line 962. I think we should do this before the err tag and check if we have hs or not.